libs/genai/langchain_google_genai/_function_utils.py

```diff
@@ -47,29 +42,36 @@ class _ToolDict(TypedDict):
     function_declarations: Sequence[_FunctionDeclarationLike]
 
 
+GeminiToolLike = Union[
+    GoogleTool,
+    _ToolDict,
+    _FunctionDeclarationLike,
+]
+
+
 def convert_to_genai_function_declarations(
-    tool: Union[
-        GoogleTool,
-        _ToolDict,
-        _FunctionDeclarationLike,
-        Sequence[_FunctionDeclarationLike],
-    ],
+    tool: GeminiToolLike,
 ) -> GoogleTool:
     if isinstance(tool, GoogleTool):
         return cast(GoogleTool, tool)
-    if isinstance(tool, type) and issubclass(tool, BaseModel):
+    if isinstance(tool, (BaseTool, Type[BaseModel], dict, FunctionDeclaration)):
         return GoogleTool(function_declarations=[_convert_to_genai_function(tool)])
     if callable(tool):
-        return _convert_tool_to_genai_function(callable_as_lc_tool()(tool))
-    if isinstance(tool, list):
-        return convert_to_genai_function_declarations({"function_declarations": tool})
+        return GoogleTool(
+            function_declarations=_convert_tool_to_genai_function(
+                callable_as_lc_tool()(tool)
+            )
+        )
+    # can not reach this code?
     if isinstance(tool, dict) and "function_declarations" in tool:
```

Comment:
This should probably be before line # 57 :)

Reply:
@lkuligin 
I think changed the order callable -> list -> dict.
Anyway, I improved code(also pytest passed). Please check again?